lib/absinthe/subscription/supervisor.ex

```diff
@@ -24,22 +24,43 @@ defmodule Absinthe.Subscription.Supervisor do
     pool_size = Keyword.get(opts, :pool_size, System.schedulers_online() * 2)
     compress_registry? = Keyword.get(opts, :compress_registry?, true)
 
-    Supervisor.start_link(__MODULE__, {pubsub, pool_size, compress_registry?})
+    document_storage =
+      Keyword.get(opts, :document_storage, Absinthe.Subscription.DefaultDocumentStorage)
+
+    storage_opts =
+      case document_storage do
+        Absinthe.Subscription.DefaultDocumentStorage ->
+          [
+            keys: :duplicate,
+            partitions: System.schedulers_online(),
+            compressed: compress_registry?
+          ]
+
+        _ ->
+          Keyword.get(opts, :storage_opts, Keyword.new())
+      end
```

Comment:
Nitpick, nonblocking:
I'm not sure about my suggestion here, but I feel like determining these storage opts should be from a function exposed through the `DocumentStorage` behavior. 

Also, I do see that `:storage_opts` can be passed in as part of the options argument for this function, so perhaps that can be accessed and determined there -- is this supervisor exclusively called by clients or would we have opportunity to control some of that within the library?

Reply:
There is still the opportunity to define and change things from the `child_spec` function since it accepts an `opts` parameter. And also an implementation is some sort of genserver for instance, those would get passed to the `start_link` function and then to `init`. I would assume these would be options you'd want throughout the life of the process.